dll/win32/shell32/iconcache.cpp

```diff
@@ -881,13 +883,26 @@ UINT WINAPI ExtractIconExW(LPCWSTR lpszFile, INT nIconIndex, HICON * phiconLarge
         PrivateExtractIconExW = (UINT(WINAPI*)(LPCWSTR,int,HICON*,HICON*,UINT)) GetProcAddress(hUser32, "PrivateExtractIconExW");
 
         if (!PrivateExtractIconExW)
-        return 0;
+        return ret;
     }
 #endif
 
     TRACE("%s %i %p %p %i\n", debugstr_w(lpszFile), nIconIndex, phiconLarge, phiconSmall, nIcons);
+    ret = PrivateExtractIconExW(lpszFile, nIconIndex, phiconLarge, phiconSmall, nIcons);
+
+    /*
+     * PrivateExtractIconEx and the internal routine, ICO_ExtractIconEx, return -1 (the unsigned integer being 0xFFFFFFFF)
```

Comment:
PrivateExtractIconEx is exported and therefore describes the contract. The internal user32 function is an implementation detail that's not relevant to shell32, so it doesn't need to be mentioned here.